addons/cookoff/functions/fnc_detonateAmmunitionServer.sqf

```diff
@@ -0,0 +1,175 @@
+#include "..\script_component.hpp"
+/*
+ * Author: Glowbal, johnb43
+ * Detonates ammunition from an object (e.g. vehicle or crate) until no ammo is left.
+ *
+ * Arguments:
+ * 0: Object <OBJECT>
+ * 1: Destroy when finished <BOOL>
+ * 2: Source <OBJECT>
+ * 3: Instigator <OBJECT>
+ *
+ * Return Value:
+ * None
+ *
+ * Example:
+ * [cursorObject, true, player, player] call ace_cookoff_fnc_detonateAmmunitionServer
+ *
+ * Public: No
+ */
+
+if (!isServer) exitWith {};
+
+params ["_object", "_destroyWhenFinished", "_source", "_instigator"];
+
+if (isNull _object) exitWith {};
+
+(_object getVariable QGVAR(cookoffMagazines)) params ["_magazines", "_totalAmmo"];
+
+private _hasFinished = _totalAmmo <= 0 || {_magazines isEqualTo []};
+
+// If the cook-off has finished or been interrupted, clean up the effects for boxes (no vehicle effects)
+if (_hasFinished ||
+    {underwater _object} || {
+    if (GVAR(ammoCookoffDuration) == 0) exitWith {true};
+
+    if (_object isKindOf "ReammoBox_F") exitWith {
+        !(GVAR(enableAmmobox) && {_object getVariable [QGVAR(enableAmmoCookoff), true]})
+    };
+
+    !(GVAR(enableAmmoCookoff) && {_object getVariable [QGVAR(enableAmmoCookoff), true]})
+}) exitWith {
+    if (_object isKindOf "ReammoBox_F") then {
+        [QGVAR(cleanupEffects), _object] call CBA_fnc_globalEvent;
+
+        // Reset variable, so the box can cook-off again
+        _object setVariable [QGVAR(isCookingOff), nil, true];
+    };
+
+    // Reset variables, so the object can detonate its ammo again
+    _object setVariable [QGVAR(cookoffMagazines), nil];
+    _object setVariable [QGVAR(isAmmoDetonating), nil, true];
+
+    // If done, destroy the object if necessary
+    if (_hasFinished && _destroyWhenFinished) then {
+        _object setDamage [1, true, _source, _instigator];
+    };
+};
+
+private _magazineIndex = floor random (count _magazines);
+private _magazine = _magazines select _magazineIndex;
+_magazine params ["_magazineClassname", "_ammoCount", "_spawnProjectile"];
+
+// Make sure ammo is at least 0
+_ammoCount = _ammoCount max 0;
```

Comment:
Is this really correct ? "at least 0" with a `max 0` statement.

Reply:
Yes it is.

This is code from #9758 though, so ignore the cookoff component.